src/battle_message.c

```diff
@@ -3538,6 +3538,12 @@ u32 BattleStringExpandPlaceholders(const u8 *src, u8 *dst, u32 dstSize)
             case B_TXT_TRAINER1_NAME: // trainer1 name
                 toCpy = BattleStringGetOpponentNameByTrainerId(gTrainerBattleOpponent_A, text, multiplayerId, GetBattlerAtPosition(B_POSITION_OPPONENT_LEFT));
                 break;
+            case B_TXT_TRAINER1_NAME_WITH_CLASS: // trainer1 name with trainer class
+                StringCopy(text, BattleStringGetOpponentClassByTrainerId(gTrainerBattleOpponent_A));
+                StringAppend(text, gText_Space2);
+                StringAppend(text, BattleStringGetOpponentNameByTrainerId(gTrainerBattleOpponent_A, text, multiplayerId, GetBattlerAtPosition(B_POSITION_OPPONENT_LEFT)));
+                toCpy = text;
```

Comment:
It doesn't matter too much, but I think this would be more efficient if written like:
```suggestion
                toCpy = text;
                text = StringCopy(text, BattleStringGetOpponentClassByTrainerId(gTrainerBattleOpponent_A));
                text = StringAppend(text, gText_Space2);
                text = StringAppend(text, BattleStringGetOpponentNameByTrainerId(gTrainerBattleOpponent_A, text, multiplayerId, GetBattlerAtPosition(B_POSITION_OPPONENT_LEFT)));
```
As-is, `StringAppend` has to scan all of `text` until it finds the `EOS` character. The change here is that `StringCopy` and `StringAppend` return a pointer to the `EOS`, so no scanning is necessary.

Reply:
Ah I see, `text` is an array atm. You'd need to have a pointer which initially points to `&text[0]` (i.e. the start) and use that everywhere I have `text`.

Reply:
Alright, done. If it looks correct.